### PR TITLE
React Calculator App Milestone 4

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
- Added logic directory inside src.
- Added calculate.js and operate.js functions and their implementation.
- Fixed linter errors.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import Big from 'big.js';
+
+const operate = (numberOne, numberTwo, operation) => {
+  switch (operation) {
+    case '-':
+      Big(numberOne - numberTwo);
```
Comment (addod19) [inline src/logic/operate.js:6]: Kindly use `minus` instead of `-`

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      break;
+    case '+':
+      Big(numberOne + numberTwo);
```
Comment (addod19) [inline src/logic/operate.js:9]: Kindly use the `plus` method instead of `+`

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      break;
+    case 'X':
+      Big(numberOne * numberTwo);
```
Comment (addod19) [inline src/logic/operate.js:12]: Kindly use the `times` method instead of `*`

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      break;
+    case '/':
+      Big(numberOne / numberTwo);
```
Comment (addod19) [inline src/logic/operate.js:15]: Kindly use the `div` method instead of `/`

File: src/logic/operate.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      break;
+    case '%':
+      Big(numberOne % numberTwo);
```
Comment (addod19) [inline src/logic/operate.js:18]: Kindly use the `mod` method instead of `%` to convert a number into a percentage

File: src/logic/calculate.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const calculate = (data, buttonName) => {
+  let { total, next } = data;
+  if (buttonName === '+/-') {
+    total *= -1;
+    next *= -1;
+  }
+  return { total, next };
+};
+
+export default calculate;
```
Comment (addod19) [inline src/logic/calculate.js]: The calculate function is not complete, kindly write a condition for all the remaining buttons; for example; `AC` to clear the input, `%` to turn a number into a percentage, and a default condition. Remember to import the operate function

File: src/logic/calculate.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const calculate = (data, buttonName) => {
+  let { total, next } = data;
```
Comment (addod19) [inline src/logic/calculate.js]: The calculator data object contains three properties: total, next, operation. Let us kindly fix this

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @SajjadAhmad14, great job on this milestone :smile: The right linter files are used :muscle: The PR and README are descriptive :+1: Commit messages are easy to understand :1st_place_medal: However, there a few comments we need to address, kindly check the inline comments below:


Once these are addressed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/logic/operate.js (hunk 1 above)
Comment (SajjadAhmad14) [reply src/logic/operate.js:6]: Thank you! Where does minus come from? Is it a built-in React or JavaScript method? I don't think so it's a requirement.

File: src/logic/calculate.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import operate from './operate'; // eslint-disable-line
+
+const calculate = (data, buttonName) => {
+  let { total, next, operation } = data;
+  operation = buttonName;
+  switch (buttonName) {
+    case '+/-':
```
Comment (afanimartin) [inline src/logic/calculate.js:10]: - Add `AC` button for clearing inputs

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work making the requested changes.
Please add the `AC` button for clearing inputs.


### Martin Afani
#### Slack: whiz25

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**